packages/mdc-textfield/mdc-text-field.scss

```diff
@@ -127,7 +127,8 @@
 
   // Vertically center aligns the text input placeholder and value for only filled variant.
   .mdc-text-field--no-label:not(.mdc-text-field--outlined):not(.mdc-text-field--textarea) & {
```

Comment:
I believe if you add `.mdc-text-field--fullWidth ...` you will also fix https://github.com/material-components/material-components-web/issues/4640